Added Behavioral Health Survey Component

Bellex0 opened the pull request:
fixes #264 

Made a survey for behavioral component. On SurveyPage4.jsx
It contains 5 question and a rechart (npm install rechart) which displays the progress for the past few days. Currently both questions and the chart is rendered in the same page. It is right after the surveypage3. Also the current data for the chart is a sample data. 
Future edits: based on feedback:
1) initially we had the questions in a multi step form (one question at a time) but in order to math the flow of the health survey, the 5 behavioral questions are rendered on the same page. We can edit it based on the feedback we can get. 
2) the chart can also be loaded on a different page, so that the participant is not influenced by his/her previous progress. 

Furthermore: in response to the issue #267  for backend integration, we are in the midst of figuring out. 
1) how to send the data from these 5 questions to the backend:
2) how to fetch data from the previous entries in the backend to display on the chart. 
Please let me know if you have any questions, comments or suggestions. 

**Preliminary Changes for Backend**
-in reducers/survey.js --> added TO_SURVEY_PAGE_3 and SET_SURVEY_PAGE_4
-actions/actions.js --> added  toSurveyPage3, setSurveyPage4,
-observation.js --> under nonphysical, added interest, sadness, sleep, energy, appetite. 

### Test plan (required)

Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

<!-- Make sure tests pass on both Travis and Circle CI. -->

### Final Checklist

- [ ] Have you bumped the version in `package.json`?
    - Second decimal for major change, third decimal for minor change. This can go past 10 i.e. 1.0.9 !=> 1.1.0, 1.0.9 => 1.0.10
    - [Read here for more info](https://semver.org/)
- [ ] Have you added any new tests necessary?
- [x ] Is your PR rebased off the most current master?
- [x ] Have you squashed all commits, or if you will merge will you squash all commits?
- [ ] Did you use yarn, not npm?
- [ x] Did you use Material-UI wherever possible?
- [ x] Did you format according to Prettier?
- [ x] Did you run all of your most recent changes locally to make sure everything is working?

SomeMoosery commented on client/src/components/behavior/chart/Rechart.js:
```diff
@@ -0,0 +1,167 @@
+  }
+
+
+//   this.setState(prevState => {
+//     return {
+//         selectedQ: [ ...prevState.selectedQ, r.answer ]
+//     }
+// })
```
remove comment

SomeMoosery commented on client/src/components/behavior/chart/Rechart.js:
```diff
@@ -0,0 +1,167 @@
+  createLine = (type, colour) => (
+    <Line 
+      activeDot={{ 
+        // onMouseUp: this.onMouseUpHandler,
```
remove comment

SomeMoosery commented on client/src/components/behavior/chart/Rechart.js:
```diff
@@ -0,0 +1,167 @@
+    
+    return (
+      <>
+      {/* <FormGroup row> */}
```
remove comment

SomeMoosery commented on client/src/components/behavior/chart/Rechart.js:
```diff
@@ -0,0 +1,167 @@
+            {sleep}
+            {energy}
+            {appetite}
+            {/* {this.createLine('Interest', 'green')} */}
+            {/* {this.createLine('Sadness', 'black')} */}
+            {/* {this.createLine('Sleep', 'purple')} */}
+            {/* {this.createLine('Energy', 'red')} */}
+            {/* {this.createLine('Appetite', 'magenta')} */}
```
remove comment

SomeMoosery commented on client/src/components/behavior/chart/Rechart.js:
```diff
@@ -0,0 +1,167 @@
+        }
+        label="Appetite"
+      />
+      {/* </FormGroup> */}
```
remove comment

SomeMoosery commented on client/src/components/survey/SurveyPage3.jsx:
```diff
@@ -58,6 +58,43 @@ const SurveyPage3 = props => {
+  const surveyPage3 = {
+    openComment
+  }
+  // const SurveyPage4 = props => {
+  //   const { setSurveyPage4, interest, sadness, sleep, energy, appetite } = props
+  //   const [interestAnswer, setInterestAnswer] = useState(interest)
+     
+  //   const [sadAnswer, setSadAnswer] = useState(sadness)
+    
+  //   const [sleepAnswer, setSleepAnswer] = useState(sleep)
+    
+  //   const [energyAnswer, setEnergyAnswer] = useState(energy)
+    
+  //   const [appetiteAnswer, setAppetiteAnswer] = useState(appetite)
+  
+
+  // const surveyPage4 = {
+  //   interestAnswer,
+  //   sadAnswer,
+  //  sleepAnswer,
+  //   energyAnswer,
+  //   appetiteAnswer
+  // };
+
+
+  const openSurveyPage4 = () => {
+    // survey.nonPhysical.openComment = openComment;
+    // submitSurvey(survey, userSession);
+    // setSurveyPage3({
+    //   openComment,
+    // });
```
remove all comments

SomeMoosery commented on client/src/components/survey/SurveyPage3.jsx:
```diff
@@ -80,6 +117,9 @@ const SurveyPage3 = props => {
             <Button onClick={submitSurveyPage3} color="secondary" className={classes.continueButton}>
               SUBMIT
             </Button>
+            <Button onClick={openSurveyPage4} color="secondary" className={classes.continueButton}>
+              CONTINUE TO BEHAVIORAL SURVEY
+            </Button>
```
Should we not just have this be an additional page in our survey i.e. have page 3 -> page 4 and then move the SUBMIT option to page 4? This could potentially confuse the user

Bellex0 replied:
I wanted to wait for the feedback as to confirm whether plugging in the behavioral survey at this page 3 is optimal. Since people who know UX might want to change page 3 directly to behavioral component. So that is why I left the button there. We can remove it once it has been tested?

SomeMoosery replied:
I'm not sure I follow - we should be sure this is working already before merging in any changes. We should loop in @AdhamAH or @BrianHHough but I think going through the full survey, so now pages 1->4, having the submit button only on page 4, would be optimal.

Or, maybe simply reorder the pages by renaming them, so that page 3 is this behavioral section, and page 4 is the "additional comments + submit"

SomeMoosery commented on client/src/components/survey/SurveyPage4.jsx:
```diff
@@ -0,0 +1,366 @@
+
+const SurveyPage4 = props => {
+const { survey, setSurveyPage4, submitSurvey, toSurveyPage3 } = props;
+// const { nonPhysical } = survey;
```
remove comment

SomeMoosery commented on client/src/components/survey/SurveyPage4.jsx:
```diff
@@ -0,0 +1,366 @@
+         appetiteAnswer
+        }
+        
+        // submitSurvey(survey, userSession);
```
remove comment

SomeMoosery commented on client/src/components/survey/SurveyPage4.jsx:
```diff
@@ -0,0 +1,366 @@
+
+ return (
+     <>
+     {/* <div>hellp</div> */}
```
remove comment

SomeMoosery commented on client/src/components/survey/SurveyPage4.jsx:
```diff
@@ -0,0 +1,366 @@
+       </>
+ )
+}
+//   );
+// };
```
remove comment

SomeMoosery commented on client/src/components/survey/SurveyPage4.jsx:
```diff
@@ -0,0 +1,366 @@
+  };
+
+export default connect(mapStateToProps, mapDispatchToProps)(SurveyPage4)
+// export default SurveyPage4
+
+// const submitSurveyPage2 = async () => {
+//     if (
+//       fever === -1 ||
+//       shortnessOfBreath === '' ||
+//       chills === '' ||
+//       dryCough === '' ||
+//       chestPain === '' ||
+//       fatigue === '' ||
+//       soreThroat === '' ||
+//       bluish === ''
+//     ) {
+//       setOpen(true);
+//     } else {
+//       const surveyPage2 = {
+//         fever,
+//         shortnessOfBreath,
+//         chills,
+//         dryCough,
+//         chestPain,
+//         fatigue,
+//         soreThroat,
+//         bluish,
+//       };
+
+//       setSurveyPage2(surveyPage2);
+//     }
+//   };
```
remove comment

SomeMoosery commented on client/src/components/behavior/chart/Rechart.js:
```diff
@@ -0,0 +1,167 @@
+      appetite: true
+
+    };
+    console.log()
```
remove `console.log()`

SomeMoosery commented on client/src/components/behavior/chart/Rechart.js:
```diff
@@ -0,0 +1,167 @@
+export default class Chart extends Component {
+  constructor() {
+    super();
+    this.state = {
+      rechartData: [
+        {name: 'Day 1', Interest: 3, Sadness: 4, Sleep: 3, Energy:4, Appetite: 5},
+        {name: 'Day 2', Interest: 6, Sadness: 6, Sleep: 5, Energy:7, Appetite: 3},
+        {name: 'Day 3', Interest: 3, Sadness: 5, Sleep: 4, Energy:2, Appetite: 3},
+        {name: 'Day 4', Interest: 2, Sadness: 2, Sleep: 4, Energy:1, Appetite: 4},
+        {name: 'Day 5', Interest: 2, Sadness: 1, Sleep: 5, Energy:5, Appetite: 3},
+        {name: 'Day 6', Interest: 5, Sadness: 2, Sleep: 3, Energy:1, Appetite: 5},
+        {name: 'Day 7', Interest: 1, Sadness: 3, Sleep: 1, Energy:5, Appetite: 3},
+      ],
+      isActive: false,
+      interest: true,
+      sadness: true,
+      sleep: true,
+      energy: true,
+      appetite: true
```
We should use React hooks `useState` here

SomeMoosery commented on client/src/components/behavior/chart/Rechart.js:
```diff
@@ -0,0 +1,167 @@
+
+import { ResponsiveContainer, Line, LineChart, Tooltip, Legend, YAxis, XAxis, CartesianGrid } from 'recharts';
+
+export default class Chart extends Component {
```
We Should use functional components - we also already have a component named `<Chart />` so this needs to be exported as something else.

Also, perhaps a clearer component name - I'm not sure what Rechart implies here

Bellex0 replied:
I changed it to BehavioralChart

SomeMoosery commented on client/src/components/behavior/chart/Rechart.js:
```diff
@@ -0,0 +1,167 @@
+    console.log(evt)
+    const targetName = evt.target.name
+    console.log(targetName)
+  this.setState({
+      [targetName]: !this.state.interest,
+      [targetName]: !this.state.sadness,
+      [targetName]: !this.state.sleep,
+      [targetName]:!this.state.energy,
+      [targetName]: !this.state.appetite
+    })
+  }
```
Similar to above, use React hooks `useState`

SomeMoosery commented on client/src/components/behavior/chart/Rechart.js:
```diff
@@ -0,0 +1,167 @@
+  )
+
+  render() {
+    const interest = this.state.interest ? this.createLine('Interest', 'green') : null 
+    const sadness = this.state.sadness ? this.createLine('Sadness', 'black') : null 
+    const sleep = this.state.sleep ? this.createLine('Sleep', 'purple') : null
+    const energy = this.state.energy ? this.createLine('Energy', 'red') : null
+    const appetite = this.state.appetite ? this.createLine('Appetite', 'magenta') : null
```
`useState`

Bellex0 commented:
1) Fixed the package.json version
2) I'm not sure how to fix the merge conflict in SurveyPage3. My local repo looks different than what is shown on the conflict. In my local repo, I don't have lines 99-119, which are in the conflict.
3) When I try to run `yarn run lint`, I get this error in the terminal: 
Oops! Something went wrong! :(
ESLint: 6.8.0.
ESLint couldn't find the config "airbnb" to extend from. Please check that the name of the config is correct.
The config "airbnb" was referenced from the config file in "/Users/belle/Desktop/practice/Corona-tracker/client/.eslintrc.json".

I checked the .eslintrc.json file, and I do have "airbnb". I also have the correct versions of "eslint": "6.8.0", "eslint-config-airbnb": "^18.1.0", "eslint-config-prettier": "^6.10.1".  So, not sure what the issue is.

SomeMoosery commented:
Do you have the most current master pulled in from your upstream? And you've ran `rebase`? If not, check out the `CONTRIBUTING.md` for guidance there. Also, note that another change was merged in, so you'll need to pull in the new master as it was updated ~20 mins ago.

In terms of the linter, have you run `yarn install` in `/client`? It may be that since ESLint is a dev dependency (not required to build/run the app), you haven't installed it.

SomeMoosery commented on client/src/components/behavior/chart/BehavioralChart.js:
```diff
@@ -0,0 +1,145 @@
+      const [appetite, setAppetite]= useState(true)
+  
+   const handleChange = evt =>  {
+    console.log(evt)
+    const targetName = evt.target.name
+    console.log(targetName)
```
remove `console.logs`

SomeMoosery commented:
Hey @Bellex0 - just checking in, how is this coming along? It would be great if we could have these changes in by Friday.

Bellex0 commented:
@SomeMoosery Hey, I'm on the last step of this component, which is trying to integrate the behavioral health survey data into the behavioral chart. I am planning to get this done ASAP!

Bellex0 commented:
We have completed the behavioral survey and the integration of the behavioral data onto the chart.  
1. In SurveyPage3, changed the "Submit" button to "Continue" button to move on to page 4 (which is the behavioral part). 
2. In SurveyPage4, implemented behavioral survey(5 questions) and added Submit button. The submit button saves the data and Date.now into the backend. 
3. Behavioral chart is incorporated in SurveyPage4. Behavioral survey data is fetched from the backend and the chart displays all of user's previous answers.
Please see this [gif](https://i.giphy.com/media/IdCI6i00pp2KaNOlzw/giphy.mp4) We can incorporate any changes based on feedback.

SomeMoosery commented:
This looks really great! Just two discussion points from the .gif - 

1. I think we should have "Back" and "Continue" on page 3 on the same row, below the text input section

2. This chart seems like it may be better fitting in the "Show Me More" section - could we maybe add it under the pre-existing chart there?

What do you / other reviewers think?

SomeMoosery commented on client/src/components/survey/SurveyPage1.jsx:
```diff
@@ -156,6 +156,10 @@ const SurveyPage1 = props => {
     }
   };
 
+  // const random = () => {
+  //   console.log(surveyPage1)
+  // }
```
Remove comment

SomeMoosery commented on client/src/components/survey/SurveyPage4.jsx:
```diff
@@ -0,0 +1,356 @@
+  const { interest, sadness, sleep, energy, appetite } = props;
+  const history = useHistory();
+  const { userSession } = useBlockstack();
+  const [interestAnswer, setInterestAnswer] = useState(interest);
+
+  const [sadAnswer, setSadAnswer] = useState(sadness);
+
+  const [sleepAnswer, setSleepAnswer] = useState(sleep);
+
+  const [energyAnswer, setEnergyAnswer] = useState(energy);
+
+  const [appetiteAnswer, setAppetiteAnswer] = useState(appetite);
```
A little pedantic, but since there's more things to change - if you could just remove the spaces between these that would be great. Just to keep things semi-consistent between files

SomeMoosery commented on client/src/components/survey/SurveyPage4.jsx:
```diff
@@ -0,0 +1,356 @@
+            </Grid>
+          </Grid>
+        </Grid>
+        <div style={{ display: 'flex', marginTop: 50, justifyContent: 'flex-end' }} />
```
No inline-styles, should be in `useStyles`

ransing commented:
> This looks really great! Just two discussion points from the .gif -
> 
> 1. I think we should have "Back" and "Continue" on page 3 on the same row, below the text input section
> 2. This chart seems like it may be better fitting in the "Show Me More" section - could we maybe add it under the pre-existing chart there?
> 
> What do you / other reviewers think?

Regarding point 2 "Show me more", The reason/s the behavioral chart was in the same page as the behavioral form was because the user can see their previous progress while entering today's data. Also "show me more" has a chart already and I am not sure if it will be crowded. I am also tryin to imagine how it will be on mobile. But I would await feedback from others as well.

Bellex0 commented:
Got rid of the inline styles and added to useStyles. Fixed the spacing and deleted console.logs. Fixed the SurveyPage3 buttons so they're in a row. Bumped package.json. 

In regards to the unit tests, we are currently unaware of the testing requirement. We are looking it up but if there are any docs that we could follow, please let us know.

acthelemann commented:
One more comment that I'd like others' thoughts on:

I feel that a higher behavior rating should be a good thing. In the current state, a higher rating is a bad thing.

ransing commented:
> One more comment that I'd like others' thoughts on:
> 
> I feel that a higher behavior rating should be a good thing. In the current state, a higher rating is a bad thing.

That is a good point, 

> One more comment that I'd like others' thoughts on:
> 
> I feel that a higher behavior rating should be a good thing. In the current state, a higher rating is a bad thing.

We wish to keep things standardized based on peer reviewed research hence that scale. I will update the docs to reflect this. To encourage user engagement, the idea is to provide individual insights based on published research.

SomeMoosery commented:
Would it be possible to present the UI on the scale @acthelemann suggests for a friendlier UX, and store this data internally as you suggest in order to remain standardized to peer-reviewed research?

So just move the same deviation from the mean, and store the value as `5-<selected value>` in Gaia

ransing commented:
> Would it be possible to present the UI on the scale @acthelemann suggests for a friendlier UX, and store this data internally as you suggest in order to remain standardized to peer-reviewed research? While we want to stay standardized with research in order to stay legitimate, we have to keep UX in mind first, as this will be primarily be used by "everyday" folk and not those with medical insights.
> 
> Meaning we move the same deviation from the mean, and store the value as `5-<selected value>` in Gaia?

So that is technically possible. Just to reiterate, the value stored on the slider is 2 instead of 4? 
`{value: 2,label: 4}` instead of `{value: 2,label: 2`}
I am unfamiliar with gaia hence won't be able to manipulate on the backend. Although I am super interested with playing around with data on the backend.

Bellex0 commented:
We implemented changes @acthelemann requested regarding cleaning up the code. Also, we incorporated checkboxes for the fever chart and the behavioral chart in Table.jsx component (when you click on Show Me More).  However, we are having some CSS issues with the fever chart rendering nicely. The bottom part of the chart gets cut off. The behavioral chart renders fine.  If someone can quickly look into the fever chart CSS issue, that'd be great.
Please this [gif](https://i.giphy.com/media/Rh4RbekK9c0753v2WX/giphy.mp4)

In regards to the rating scale, I understand that we should keep the rating scale consistent for all questions. However, the way these questions are phrased, it doesn’t sound rational to the user if 5 means “I’m feeling great, I haven’t lost any interest” and 1 means  “I’ve had significant loss of interest.” User will naturally equate 5 with being the most and 1 with none.  
But if there is a strong preference for changing the scale, then we can change it.

SomeMoosery commented:
Yeah I would say in terms of sizing/styling charts let's move that into a different issue so this PR doesn't go out of scope, which it's starting to a bit already now.

kevin-Price commented:
We should definitely sync up on this on discoed. I’m also working on a line
chart for the show more page. Would be good to have the same style of chart
across the app.

I’ll reply in a bit with my repo and the code for that.



On Wed, Apr 15, 2020 at 7:40 AM Carter Klein <notifications@github.com>
wrote:

> Yeah I would say in terms of sizing/styling charts let's move that into a
> different issue so this PR doesn't go out of scope, which it's starting to
> a bit already now.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/COVID-19-electronic-health-system/Corona-tracker/pull/460#issuecomment-614080125>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM2MFYWUKBIX4WIK7OJJ7STRMXBNJANCNFSM4MFXRJTQ>
> .
>

acthelemann commented on client/src/components/behavior/chart/BehavioralChart.js:
```diff
@@ -0,0 +1,139 @@
+const BehavioralChart = props => {
+  const { observations } = props;
+
+  const behaveData = observations.slice(observations.length - 7, observations.length).map(obj => {
```
We shouldn't `slice` here.

ransing replied:
So the idea was to have the last 7 days record show up. Is there a more adequate way to do it?

acthelemann replied:
Hmm okay. I haven't looked at the `rechart` docs, but is there a way to have some range on the x-axis?

ransing replied:
I can look into that. I am sure there is. 
If there is a range and let's say X-axis is limited to 7, the goal would be to show only the last 7 days. So we will have to limit the 7 records going there from the map. OR make sure the rechart only takes in the data from the last 7 entries. Also if this is not necessary feature at the moment then we can also integrate it in the next iteration. The idea is to not have the chart crowded (hence the 7 days limit).

acthelemann replied:
Yeah, this would be added in the next iteration.

Currently, users are able to submit the survey more than once, so there could be more than one submission per day.

Also, maybe the chart could be scrollable (?) or able to change the date range to see past data.

Just some ideas to keep in mind.

kevin-Price commented:
This is great! Im also using rechart for #370 . Maybe we should make that standard for charts going forward.

ransing commented:
> Style-wise, it needs adjustment for the small screens, it can be done later with a separate issue because you already did a lot! Great work!
They styles can be modified. Also also the color of the line can be changed. The lines overlap if it is the same variable. 
> The temperature charts I think @kevin-Price or @gitalink working on it, maybe they should take a look because I think this will affect what they are working on.
> The charts new toggle is great! it gives me some ideas! But why the charts still in the `SurveyPage4.jsx`, do you want to keep it there?
The charts were in both location based on feedback from this thread. 
> one thing, if there is no entered data it shows undefined
> ![Screenshot 2020-04-15 at 13 50 38](https://user-images.githubusercontent.com/55054963/79334224-28e50480-7f20-11ea-89ba-95e7201a2184.png)
> 
> I am guessing this because of the returned value from useState but I am not sure

Yes, the empty Data shows up as NaN. Validations will get rid of it.

ransing commented:
> > Style-wise, it needs adjustment for the small screens, it can be done later with a separate issue because you already did a lot! Great work!
>
>  The styles can be modified. Also also the color of the line can be changed. The lines overlap if it is the same variable.
> > The temperature charts I think @kevin-Price or @gitalink working on it, maybe they should take a look because I think this will affect what they are working on.
> > The charts new toggle is great! it gives me some ideas! But why the charts still in the `SurveyPage4.jsx`, do you want to keep it there?
>
>   The charts were in both location based on feedback from this thread.
> > one thing, if there is no entered data it shows undefined
> > ![Screenshot 2020-04-15 at 13 50 38](https://user-images.githubusercontent.com/55054963/79334224-28e50480-7f20-11ea-89ba-95e7201a2184.png)
> > I am guessing this because of the returned value from useState but I am not sure
> 
> Yes, the empty Data shows up as NaN. Validations will get rid of it.

ransing commented:
> We should definitely sync up on this on discoed. I’m also working on a line chart for the show more page. Would be good to have the same style of chart across the app. I’ll reply in a bit with my repo and the code for that.
> […](#)
> On Wed, Apr 15, 2020 at 7:40 AM Carter Klein ***@***.***> wrote: Yeah I would say in terms of sizing/styling charts let's move that into a different issue so this PR doesn't go out of scope, which it's starting to a bit already now. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#460 (comment)](https://github.com/COVID-19-electronic-health-system/Corona-tracker/pull/460#issuecomment-614080125)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AM2MFYWUKBIX4WIK7OJJ7STRMXBNJANCNFSM4MFXRJTQ> .

So we are discussing it on the channel #health-log on discord for this and the future issues

SomeMoosery commented:
While it's good to discuss minor details in Discord we should keep most pertinent discussions in the PR history so we're able to more easily trace conversations and decisions.

I think for the time being, I would suggest removing the chart entirely for this PR, merging this in, and then creating a new issue / opening a new PR for the addition of the chart. The original issue was probably too broad and leads to PRs getting this large and unwieldy.

SomeMoosery commented on client/src/components/Table.jsx:
```diff
@@ -22,38 +47,87 @@ const LogTable = props => {
+      {behaveClicked ? <Typography variant="h2">Your Behavioral Health Progress</Typography> : null}
+      {renderBehavior}
+      {behaveClicked ? (
+        <div className={classes.tagLine}>
+          <Typography variant="h6">
+            See this past week's daily progress. Hover over the chart to see individual scores.
+          </Typography>
+        </div>
```
Let's remove the chart from the UI at least for this PR - I will open up another issue capturing implementing it. Let's just capture the scope of adding behavioral elements to CoronaTracker for this PR

ransing replied:
for Table.jsx? it would be really difficult to remove the entire component because then checkboxes are integrated with it too.

SomeMoosery replied:
I wouldn't remove the entire component, just remove it from the UI - you can use empty `<div>`s  etc.

This PR is getting much too large so we need to merge in a minimum viable solution here and then scope out more pointed issues to implement the rest from here.

Bellex0 replied:
I do agree this PR is getting bigger than previously anticipated.  To clarify, we should remove the checkboxes (fever & behavioral chart) and the behavioral chart from Table.jsx?

SomeMoosery replied:
If possible, yes. They're great features to have and I'd love to to get them merged in, just in a different PR so we can keep scope a little clearer :)

kevin-Price replied:
Im also working on table.jsx for issue #370 , so it might be a good idea to cut it out of scope for 370 when you make a new issue for it

Bellex0 replied:
Removed the checkboxes and chart. Hopefully, this is good to go now!

SomeMoosery commented:
Also be sure to rebase off of current master as tests are failing

SomeMoosery commented:
Unfortunately it seems like it's still not scaling properly for mobile screens. These are screenshots for an iPhone X's dimensions:

<img width="307" alt="Screen Shot 2020-04-15 at 3 17 19 PM" src="https://user-images.githubusercontent.com/15767602/79379055-47e99380-7f2c-11ea-859a-20ecd1dff53e.png">

<img width="307" alt="Screen Shot 2020-04-15 at 3 17 26 PM" src="https://user-images.githubusercontent.com/15767602/79379058-4a4bed80-7f2c-11ea-89a1-b63816ab9b1d.png">

I would suggest removing the chart from the survey as well, adding it everywhere in a new PR once it's fixed for all screen sizes. I would just fix up the formatting here (look to see how sliders are laid out in Page 1) and then it should be good to go 👍

Bellex0 commented:
Removed chart from survey and fixed slider, added emojis.

SomeMoosery commented:
Great work on this @Bellex0 and @ransing - along with everyone who commented/suggested/requested changes. This is what open-source is all about!
